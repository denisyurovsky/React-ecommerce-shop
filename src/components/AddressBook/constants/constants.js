export const LISTBOX_PADDING = 8;
export const DOUBLE_LISTBOX_PADDING = LISTBOX_PADDING * 2;
export const LIMIT_NUMBER_ITEMS = 8;
export const ITEM_SIZE_LG = 48;
export const ITEM_SIZE_MD = 36;
