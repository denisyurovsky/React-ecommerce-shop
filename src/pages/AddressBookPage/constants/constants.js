export const MODAL_TYPE = {
  EDIT: 'edit',
  ADD: 'add',
};

export const EMPTY_ADDRESS = {
  title: '',
  name: '',
  surname: '',
  country: { name: '', id: '' },
  city: '',
  street: '',
  building: '',
  flat: '',
  phone: '',
  zip: '',
};

export const LINKS = [
  { url: '/', text: 'Home' },
  { url: '/profile', text: 'Profile' },
  {
    url: `/profile/address-book`,
    text: 'Address book',
  },
];

export const RUSSIA = 'Russian Federation';
