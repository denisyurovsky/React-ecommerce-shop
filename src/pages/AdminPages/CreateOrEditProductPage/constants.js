export const PRICE_ERROR = 'Only numbers';
export const DESCRIPTION_ERROR = 'Only latin characters';
export const NOT_FOUND_PRODUCT_ERROR =
  'The item you are looking for was not found';
