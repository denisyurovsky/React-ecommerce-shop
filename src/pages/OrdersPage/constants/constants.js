export const breadcrumbsLinks = [
  { url: '/', text: 'Home' },
  { url: '/profile', text: 'Profile' },
  { url: '/profile/orders', text: 'My orders' },
];
