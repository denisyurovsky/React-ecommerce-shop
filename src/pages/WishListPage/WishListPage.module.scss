@import "../../assets/styles/colors.scss";
@import "../../assets/styles/breakpoints.scss";

.page {
  width: 100%;
}

.emptyWishList {
  font-size: 40px !important;
  margin-top: 100px !important;
  color: $grey700;
  text-align: center;
}

.cardsContainer {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;

  @media all and (max-width: $lg) {
    justify-content: center;
  }
}
