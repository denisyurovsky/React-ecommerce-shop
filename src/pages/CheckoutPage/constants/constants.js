import { pathNames } from '../../../constants/pathNames';

export const LINKS = [
  { url: '/', text: 'Home' },
  {
    url: pathNames.CHECKOUT,
    text: 'Checkout',
  },
];

export const EMPTY_ORDER = {
  userId: null,
  id: null,
  products: [],
  status: null,
  addressId: null,
  deliveryAddress: {},
  deliveryPrice: null,
  totalPrice: null,
  totalDiscountPrice: null,
  totalQuantity: null,
  deliveredAt: null,
  createdAt: null,
  updatedAt: null,
};
