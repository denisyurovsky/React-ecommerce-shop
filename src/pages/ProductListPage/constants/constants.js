import { pathNames } from '../../../constants/pathNames';

export const sortTypes = {
  NEW_FIRST: 'new',
  OLD_FIRST: 'old',
  EXPENSIVE_FIRST: 'expensive',
  CHEAP_FIRST: 'cheap',
};

export const pageView = {
  MODULE_VIEW: 'module',
  LIST_VIEW: 'list',
};

export const sortObj = {
  [sortTypes.NEW_FIRST]: { field: 'createdAt', order: 'desc' },
  [sortTypes.OLD_FIRST]: { field: 'createdAt', order: 'asc' },
  [sortTypes.EXPENSIVE_FIRST]: { field: 'price', order: 'desc' },
  [sortTypes.CHEAP_FIRST]: { field: 'price', order: 'asc' },
};

export const BreadcrumbsLinks = [
  { url: '/', text: 'Home' },
  { url: pathNames.PRODUCTS, text: 'Products' },
];

export const NUMBER_ITEMS_ON_PAGE = 20;

export const MOBILE_WIDTH = 540;

export const defaultParams = {
  entity: 'products',
  filters: null,
  sort: sortObj[sortTypes.NEW_FIRST],
  currentPage: 1,
  itemsPerPage: NUMBER_ITEMS_ON_PAGE,
  q: '',
};
