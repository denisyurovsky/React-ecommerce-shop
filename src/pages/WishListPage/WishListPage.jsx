import { Typography, Box } from '@mui/material';
import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { toast } from 'react-toastify';

import { getProductsByIds } from '../../api/products';
import Breadcrumbs from '../../components/Breadcrumbs/Breadcrumbs';
import Card from '../../components/ProductCard/Card/Card';
import { Title } from '../../components/Title/Title';
import { notificationError } from '../../helpers/constants/constants';
import { getWishlist, selectUser } from '../../store/user/userSlice';
import { pageView } from '../ProductListPage/constants/constants';

import styles from './WishListPage.module.scss';

export const WishListPage = () => {
  const user = useSelector(selectUser);
  const [products, setProducts] = useState([]);
  const wishlist = useSelector(getWishlist);
  const isWished = (productId, wishlist) => new Set(wishlist).has(productId);

  const links = [
    { url: '/', text: 'Home' },
    { url: '/profile', text: 'Profile' },
  ];

  useEffect(() => {
    const wishlistIds = user.user.wishlist;

    const getWishListProducts = async (wishlistIds) => {
      try {
        const response = await getProductsByIds(wishlistIds);

        setProducts(response.data);
      } catch (err) {
        toast(notificationError);
      }
    };

    getWishListProducts(wishlistIds);
  }, [user.user.wishlist]);

  return (
    <div className={styles.page}>
      <Breadcrumbs links={links} />
      <Title>Wishlist</Title>
      {!products || products.length === 0 ? (
        <Typography className={styles.emptyWishList}>
          Your Wishlist is empty
        </Typography>
      ) : (
        <Box className={styles.cardsContainer}>
          {products.map((product) => (
            <Card
              key={product.id}
              product={{
                ...product,
                isAddedToWishlist: isWished(product.id, wishlist),
              }}
              cardShape={pageView.MODULE_VIEW}
            />
          ))}
        </Box>
      )}
    </div>
  );
};
