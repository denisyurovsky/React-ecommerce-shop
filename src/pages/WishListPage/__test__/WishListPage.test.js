import { configureStore } from '@reduxjs/toolkit';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import React from 'react';
import { Provider } from 'react-redux';

import { authStatus } from '../../../constants/authConstants';
import { USER_ROLE } from '../../../constants/constants';
import cartReducer from '../../../store/cart/cartSlice';
import productsReducer from '../../../store/products/productsSlice';
import userReducer from '../../../store/user/userSlice';
import testProducts from '../../../test-utils/dto/productsDto';
import { wishlistsDto } from '../../../test-utils/dto/wishlistsDto';
import renderWithStore from '../../../test-utils/renderWithStore';
import RouterConnected from '../../../test-utils/RouterConnected';
import { WishListPage } from '../WishListPage';

const { FULFILLED } = authStatus;

const serverUser = {
  id: 1,
  wishlists: wishlistsDto,
};
const initialUser = {
  user: {
    id: 1,
    wishlists: wishlistsDto,
    amountOfTries: 0,
    role: USER_ROLE.CONSUMER,
  },
  loginStatus: FULFILLED,
};

const initialCart = {
  sellers: {},
  totalPrice: 0,
  totalQuantity: 0,
};

const stateForTests = {
  user: initialUser,
  cart: initialCart,
};

const store = configureStore({
  reducer: {
    user: userReducer,
    products: productsReducer,
    cart: cartReducer,
  },
  preloadedState: stateForTests,
});

const handlersFulfilled = [
  rest.patch('/users/:userId', (req, res, ctx) => {
    return res(ctx.json(serverUser));
  }),
  rest.get('/users/1', (req, res, ctx) => {
    return res(ctx.json(serverUser));
  }),
  rest.get('/products/:productId', (req, res, ctx) => {
    const { productId } = req.params;

    return res(ctx.json(testProducts[productId]), ctx.status(200));
  }),
  rest.get('/products', (req, res, ctx) => {
    return res(ctx.json(testProducts), ctx.status(200));
  }),
  rest.get('/products/', (req, res, ctx) => {
    const response = [];

    if (req.url.searchParams) {
      req.url.searchParams.forEach((item) => {
        response.push(testProducts[item]);
      });
    }

    return res(ctx.json(response), ctx.status(200));
  }),
];

const serverFulfilled = setupServer(...handlersFulfilled);

describe('WishListPage component', () => {
  describe('snapshots', () => {
    it('renders a WishListPage snapshot with only one wishlist - default', () => {
      const { asFragment } = renderWithStore(
        <RouterConnected component={<WishListPage />} />
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('renders a WishListPage snapshot with several wishlists', async () => {
      serverFulfilled.listen();
      const { asFragment } = render(
        <Provider store={store}>
          <RouterConnected component={<WishListPage />} />
        </Provider>
      );

      expect(asFragment()).toMatchSnapshot();

      serverFulfilled.resetHandlers();
      serverFulfilled.close();
    });
  });

  describe('test actions', () => {
    beforeAll(() => serverFulfilled.listen());
    afterAll(() => serverFulfilled.close());

    it('should show product cards', async () => {
      render(
        <Provider store={store}>
          <RouterConnected component={<WishListPage />} />
        </Provider>
      );
      const expandButtons = screen.getAllByTestId('ExpandMoreIcon');

      userEvent.click(expandButtons[0]);
      expect(screen.getByRole('progressbar')).toBeInTheDocument();
      expect(
        await screen.findByText(/Intelligent Cotton Pants/i)
      ).toBeInTheDocument();
    });

    it('should delete wishlist', async () => {
      render(
        <Provider store={store}>
          <RouterConnected component={<WishListPage />} />
        </Provider>
      );
      expect(screen.getByText('Gifts for the wife')).toBeInTheDocument();
      const deleteButtons = screen.getAllByTestId('delete');

      userEvent.click(deleteButtons[0]);
      userEvent.click(screen.getByRole('button', { name: 'Yes' }));
      expect(await screen.findAllByTestId('wishlist-name')).toHaveLength(2);
    });

    it('should edit wishlist', async () => {
      render(
        <Provider store={store}>
          <RouterConnected component={<WishListPage />} />
        </Provider>
      );
      expect(screen.getByText('Default wishlist')).toBeInTheDocument();
      const editButtons = screen.getAllByTestId('edit');

      userEvent.click(editButtons[0]);
      const input = screen.getByLabelText('Wishlist Name');

      fireEvent.focus(input);
      userEvent.type(input, 'New name wishlist');
      fireEvent.blur(input);
      const wishlists = await screen.findAllByTestId('wishlist-name');

      expect(wishlists).toHaveLength(2);
      expect(wishlists[0]).toHaveTextContent('New name wishlist');
    });
  });
});
