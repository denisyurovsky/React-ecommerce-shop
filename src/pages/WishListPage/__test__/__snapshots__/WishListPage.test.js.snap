// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WishListPage component snapshots renders a WishListPage snapshot with only one wishlist - default 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="breadcrumbs"
      role="presentation"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="breadcrumb"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="breadcrumb"
              href="/users"
            >
              Users
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="breadcrumb"
              href="/profile"
            >
              Profile
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="header"
    >
      <h2
        class="header"
        data-testid="title"
      >
        Wishlist
      </h2>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg container css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root container css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          data-testid="menuList"
          id="composition-menu"
          role="menu"
          tabindex="-1"
        >
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="0"
          >
            <a
              class="link grey"
              href="/profile"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Profile
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <a
              class="link grey"
              href="/profile/wishlist"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FavoriteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Wishlist
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <a
              class="link grey"
              href="/profile/orders"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="Inventory2Icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 2H4c-1 0-2 .9-2 2v3.01c0 .72.43 1.34 1 1.69V20c0 1.1 1.1 2 2 2h14c.9 0 2-.9 2-2V8.7c.57-.35 1-.97 1-1.69V4c0-1.1-1-2-2-2zm-5 12H9v-2h6v2zm5-7H4V4h16v3z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Orders
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <a
              class="link grey"
              href="/profile/address-book"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ImportContactsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.5 4.5c-1.95 0-4.05.4-5.5 1.5-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6v14.65c0 .65.73.45.75.45C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5 1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.41.21.75-.19.75-.45V6c-1.49-1.12-3.63-1.5-5.5-1.5zm3.5 14c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5V8c1.35-.85 3.8-1.5 5.5-1.5 1.2 0 2.4.15 3.5.5v11.5z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Address book
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root profile css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="page"
        >
          <div
            class="cardsContainer MuiBox-root css-0"
          >
            <div
              class="wishlistCard MuiBox-root css-0"
              data-cy="wishlistCard"
            >
              <div
                class="header"
              >
                <button
                  aria-expanded="false"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium expandIcon css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  data-cy="expandBtn"
                  data-testid="show more"
                  expand="false"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ExpandMoreIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <div
                  class="title MuiBox-root css-0"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 name css-ahj2mt-MuiTypography-root"
                    data-testid="wishlist-name"
                  >
                    Default wishlist
                  </p>
                  <div
                    class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium button css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      data-testid="edit"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                        data-testid="EditIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;

exports[`WishListPage component snapshots renders a WishListPage snapshot with several wishlists 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="breadcrumbs"
      role="presentation"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="breadcrumb"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="breadcrumb"
              href="/users"
            >
              Users
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="breadcrumb"
              href="/profile"
            >
              Profile
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="header"
    >
      <h2
        class="header"
        data-testid="title"
      >
        Wishlist
      </h2>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg container css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root container css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          data-testid="menuList"
          id="composition-menu"
          role="menu"
          tabindex="-1"
        >
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="0"
          >
            <a
              class="link grey"
              href="/profile"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Profile
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <a
              class="link grey"
              href="/profile/wishlist"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FavoriteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Wishlist
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <a
              class="link grey"
              href="/profile/orders"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="Inventory2Icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 2H4c-1 0-2 .9-2 2v3.01c0 .72.43 1.34 1 1.69V20c0 1.1 1.1 2 2 2h14c.9 0 2-.9 2-2V8.7c.57-.35 1-.97 1-1.69V4c0-1.1-1-2-2-2zm-5 12H9v-2h6v2zm5-7H4V4h16v3z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Orders
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <a
              class="link grey"
              href="/profile/address-book"
              style="opacity: 1;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ImportContactsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.5 4.5c-1.95 0-4.05.4-5.5 1.5-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6v14.65c0 .65.73.45.75.45C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5 1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.41.21.75-.19.75-.45V6c-1.49-1.12-3.63-1.5-5.5-1.5zm3.5 14c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5V8c1.35-.85 3.8-1.5 5.5-1.5 1.2 0 2.4.15 3.5.5v11.5z"
                />
              </svg>
              <span
                class="menuTitle"
              >
                My Address book
              </span>
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root profile css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="page"
        >
          <div
            class="cardsContainer MuiBox-root css-0"
          >
            <div
              class="wishlistCard MuiBox-root css-0"
              data-cy="wishlistCard"
            >
              <div
                class="header"
              >
                <button
                  aria-expanded="false"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium expandIcon css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  data-cy="expandBtn"
                  data-testid="show more"
                  expand="false"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ExpandMoreIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <div
                  class="title MuiBox-root css-0"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 name css-ahj2mt-MuiTypography-root"
                    data-testid="wishlist-name"
                  >
                    Default wishlist
                  </p>
                  <div
                    class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium button css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      data-testid="edit"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                        data-testid="EditIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="wishlistCard MuiBox-root css-0"
              data-cy="wishlistCard"
            >
              <div
                class="header"
              >
                <button
                  aria-expanded="false"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium expandIcon css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  data-cy="expandBtn"
                  data-testid="show more"
                  expand="false"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ExpandMoreIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <div
                  class="title MuiBox-root css-0"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 name css-ahj2mt-MuiTypography-root"
                    data-testid="wishlist-name"
                  >
                    Gifts for the wife
                  </p>
                  <div
                    class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium button css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      data-testid="edit"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                        data-testid="EditIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-label="delete"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium button css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      data-testid="delete"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                        data-testid="DeleteIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
